lib/mllp/peer.ex

```diff
@@ -0,0 +1,80 @@
+defmodule MLLP.Peer do
+  require Logger
+
+  @type t :: %{
+          :transport => :ranch_tcp | :ranch_ssl,
+          :socket => :ranch_transport.socket(),
+          :client_info => {:inet.ip_address(), :inet.port_number()}
+        }
+
+  @type error_type ::
+          :client_ip_not_allowed
+          | :fail_to_verify_client_cert
+
+  @spec validate(t(), Keyword.t()) :: {:ok, :success} | {:error, error_type()}
+
+  def validate(peer, options) do
+    allowed_clients = Keyword.get(options, :allowed_clients, [])
```

Comment:
If we transformed allowed_clients into a map, then we could use `is_map_key/2 guard in the new Peer module  and avoid a conditional `if` and `else` in the body. What do you think?